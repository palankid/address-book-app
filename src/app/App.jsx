import React from 'react';
import { hot } from 'react-hot-loader/root';

import './App.scss'
import Details from '../features/details';

const App = () => {
  return (
    <Details />
  )
};

export default hot(App);
