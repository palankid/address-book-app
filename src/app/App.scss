@import '~semantic-ui-css/semantic.css';

#app {
  height: 100%;
}
