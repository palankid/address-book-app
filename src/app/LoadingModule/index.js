import LoadingModule from './LoadingModule';
import './LoadingModule.scss';

export default LoadingModule;
