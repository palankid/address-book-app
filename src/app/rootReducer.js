import { combineReducers } from 'redux-starter-kit'

import detailsReducer from '../features/details/reducer'

const rootReducer = combineReducers({
  details: detailsReducer
});

export default rootReducer
