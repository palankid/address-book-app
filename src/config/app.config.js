export const MAX_USERS_COUNT = 90;
export const PAGE_SIZE = 30;
export const SEED_NAME = 'addressbook';
