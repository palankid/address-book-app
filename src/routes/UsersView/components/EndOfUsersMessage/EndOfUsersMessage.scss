.end-of-users-message {
  text-align: center;
  margin: 20px 0;

  &__text {
    font-size: 1.5rem;
    padding: 10px;
    color: #FFFFFF;
    background-color: #000000;
  }
}
