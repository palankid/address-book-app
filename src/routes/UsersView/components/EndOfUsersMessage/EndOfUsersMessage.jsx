import React from 'react';
import { array } from 'prop-types';

import { MAX_USERS_COUNT } from '../../../../config/app.config';

const EndOfUsersMessage = ({ users }) => {
  const isMessageHidden = users.length < MAX_USERS_COUNT;
  if (isMessageHidden) { return null }

  return (
    <div className="end-of-users-message">
      <span className="end-of-users-message__text">
        End of users catalog
      </span>
    </div>
  );
};

EndOfUsersMessage.propTypes = {
  users: array
};

export default EndOfUsersMessage;
