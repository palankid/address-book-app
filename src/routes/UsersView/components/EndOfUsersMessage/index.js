import EndOfUsersMessage from './EndOfUsersMessage';
import './EndOfUsersMessage.scss';

export default EndOfUsersMessage;
