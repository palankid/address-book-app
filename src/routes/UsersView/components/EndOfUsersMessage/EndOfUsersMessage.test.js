import React from 'react';
import { shallow } from 'enzyme';

import { MAX_USERS_COUNT } from '../../../../config/app.config';

import EndOfUsersMessage from './EndOfUsersMessage';

describe('EndOfUsersMessage', () => {
  const createMockedUsers = (count) => {
    return new Array(count).map(() => ({
      prop: true
    }));
  };

  it('should not display message if max number of users is not reached', () => {
    const users = createMockedUsers(2);
    const wrapper = shallow(
      <EndOfUsersMessage
        users={users}
      />
    );

    expect(wrapper.isEmptyRender()).toBeTruthy();
  });

  it('should display message if max number of users is reached', () => {
    const users = createMockedUsers(MAX_USERS_COUNT);
    const wrapper = shallow(
      <EndOfUsersMessage
        users={users}
      />
    );

    expect(wrapper.isEmptyRender()).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });
});
