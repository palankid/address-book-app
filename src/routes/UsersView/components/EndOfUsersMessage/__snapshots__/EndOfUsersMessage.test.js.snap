// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndOfUsersMessage should display message if max number of users is reached 1`] = `
<div
  className="end-of-users-message"
>
  <span
    className="end-of-users-message__text"
  >
    End of users catalog
  </span>
</div>
`;
